### PR TITLE
Edit evaluation

### PR DESCRIPTION
**Pull request by rachaelcodes**
Add links for users to edit evaluations which they have created
* also adds a new view for editing the title and departments of an evaluation

### CODE REVIEW

File: evaluation_registry/templates/evaluation_detail.html [hunk 1]
```diff
@@ -1,17 +1,31 @@
 {% extends "base.html" %}
+{% import 'macros.html' as macros with context %}
 
 {% block content %}
   <div class="govuk-grid-row">
-    <a href="{{ url('homepage') }}?{{ request.GET.urlencode() }}" class="govuk-back-link">Back</a>
```
Comment (rachaelcodes) [inline evaluation_registry/templates/evaluation_detail.html:5]: There is a ticket to fix the routing for this, am removing it while the routing is broken.

File: evaluation_registry/evaluations/views.py [hunk 2]
```diff
@@ -527,3 +533,75 @@ def evaluation_update_cost_view(request, uuid):
...
+                form.save()
+
+                if form.cleaned_data["lead_department"] != evaluation.lead_department:
+                    lead_department_link = EvaluationDepartmentAssociation.objects.get(
+                        evaluation=evaluation, is_lead=True
+                    )
+                    lead_department_link.department = form.cleaned_data["lead_department"]
+                    lead_department_link.save()
+
+                to_add = set(form.cleaned_data["departments"]).difference(evaluation.other_departments) or []
```
Comment (gecBurton) [inline evaluation_registry/evaluations/views.py:563]:
```suggestion
                to_add = set(form.cleaned_data["departments"]).difference(evaluation.other_departments) or set()
```
this really doesnt matter....im just surprised mypy doesnt complain here

File: evaluation_registry/evaluations/views.py [hunk 3]
```diff
@@ -527,3 +533,75 @@ def evaluation_update_cost_view(request, uuid):
...
+                        evaluation=evaluation,
+                        department=department,
+                    )
+
+                to_remove = set(evaluation.other_departments).difference(form.cleaned_data["departments"]) or []
+                for department in to_remove:
+                    EvaluationDepartmentAssociation.objects.filter(
+                        evaluation=evaluation, department=department
+                    ).delete()
+
```
Comment (gecBurton) [inline evaluation_registry/evaluations/views.py:576]:
```python
 EvaluationDepartmentAssociation.objects.filter(
     evaluation=evaluation, 
     department__in=evaluation.other_departments,
).exclude(
     department__in=form.cleaned_data["departments"],
).delete()
```
could get the ORM to most of the work here?

**Comment by nmenezes0** [comment]
Merging so changes can be deployed to dev (can make any cleaning-up changes on another PR).

**Review by nmenezes0 — APPROVED** [review]

**Comment by Nick-babs** [comment]
A slight problem is because the users when they go to the edit my evaluation page there is no way on either of the flows for the user to continue the journey to the confirmation screen. A change needs to be made to make the 'Confirm sharing the evaluation' page able to be edited

**Pull request merged**